experimental/pkg/apis/mci/v1alpha1/types.go

```diff
@@ -0,0 +1,117 @@
+// Copyright 2019 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	core "k8s.io/api/core/v1"
+	extensions "k8s.io/api/extensions/v1beta1"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// MultiClusterIngress is the API resource used to create an Ingress spanning
+// multiple clusters.
+// The API schema here is very similar to a single cluster Ingress resource
+// with the primary difference being that the service name referenced in the
+// path refers to a MultiClusterService resource rather than a Service object.
+// +k8s:openapi-gen=true
+type MultiClusterIngress struct {
+	meta_v1.TypeMeta   `json:",inline"`
+	meta_v1.ObjectMeta `json:"metadata,omitempty"`
+	Spec               MultiClusterIngressSpec   `json:"spec"`
+	Status             MultiClusterIngressStatus `json:"status,omitempty"`
+}
+
+// MultiClusterIngressSpec is the spec for a MultiClusterIngress.
+type MultiClusterIngressSpec struct {
+	// Template contains the ingress spec used to create a
+	// MultiClusterIngress.
+	Template IngressSpecTemplate `json:"template,omitempty"`
+}
+
+// MultiClusterIngressStatus is the status of a MultiClusterIngress.
+type MultiClusterIngressStatus struct {
+	// VIP is the virtual IP address of the load balancer.
+	VIP string
```

Comment:
1. Can `VIP` be an array?
2. Is it possible that Service created in multiple clusters have provisioned a loadbalancers with different IP. How that can be handled or avoided?